  # Contributors 
    Primary contributor: Maria Nikoghosyan
    Contributing authors: Nate Zadirako, Emma Hovhannisyan

## Session Schedule

| Date (2025)   | Time          | Topic                                   | Folder                                                              |
| ------------- | ------------- | --------------------------------------- | ------------------------------------------------------------------- |
| **13 August**   | 11:00 – 12:00 | Lecture: Genomic Variations     | [`Genetic variations/`](./)                                      |
|   **13 August**  | 12:00 – 13:00   | Practice: Variant Calling   | [`varinat_calling/`](./varinat_calling/)                                      |
| **13 August**   | 14:00 – 15:00   | Practice: VCF File Manipulation    | [`vcf_file_manipulation/`](./vcf_file_manipulation/)  |
| **14 August** | 11:00 – 12:00 | Lecture: Integrative Genomics Viewer (IGV)| [`Integrative_Genomics_Viewer_(IGV)/`](./Integrative_Genomics_Viewer_(IGV)/) |
| **14 August** | 12:00 – 13:00 |Practice: Integrative Genomics Viewer (IGV)| [`Integrative_Genomics_Viewer_(IGV)/`](./Integrative_Genomics_Viewer_(IGV)/)  |


---

  # Content 
Genomic variations are the differences in DNA sequence—such as single-nucleotide polymorphisms (SNPs), insertions, deletions, and structural rearrangements—that distinguish individuals or populations and underlie traits, evolution, and disease. Variant calling is the computational process of comparing high-throughput sequencing reads to a reference genome to detect and catalog these variations with quality metrics. Accurate variant calling transforms raw sequencing data into actionable biological insights for research and clinical applications.

### Lecture
[Here](https://docs.google.com/presentation/d/1iBj2Rdx6Ue0G7pYThBh_7XcEZjCzS00m/edit?slide=id.p1#slide=id.p1) you will find the slides on genomic variation and variant calling 

[Here]() you will find the slides on how to visualize NGS data using the Integrative Genomics Viewer (IGV).

### Practice

In the **variant_calling** folder, you will find the materials for the practical session on variant calling.

- **variant_calling_pipeline.md** — a step-by-step guide to the variant-calling pipeline. All file paths are specified for their locations on the server where you will execute the commands.

In the **vcf_file_manipulation** folder, you will find the materials for the practical session related to VCF file manipulations.



  # TODO
     Adjust paths 
