---
title: "OMICSS25:Practice of the Transcriptomics Course"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Welcome to the Practice of the Transcriptomics Course

A basic step in analyzing RNA-seq count data is finding genes that are expressed differently between conditions. The data usually come in a table showing, for each sample, how many sequencing reads map to each gene. Similar types of data also come from experiments like ChIP-Seq, HiC, shRNA screens, and mass spectrometry. An important question in analysis is to measure and test if there are real differences between conditions, beyond the normal variability seen within each condition. The DESeq2 package helps test for differential expression using negative binomial models. It uses data-based prior distributions to improve the estimates of variability and log fold changes.

# System preparation
## Package installation

Please follow the provided instructions to install "DESeq2", "EnhancedVolcano" and "ashr" packages.

To install these packages, start R and enter:
```r
if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

  BiocManager::install("DESeq2")
  BiocManager::install("EnhancedVolcano")
options(repos = c(CRAN = "https://cran.r-project.org"))
install.packages('ashr')
```

```{r packages, include=FALSE}
  if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")

  BiocManager::install("DESeq2")
  BiocManager::install("EnhancedVolcano")
  options(repos = c(CRAN = "https://cran.r-project.org"))
  install.packages('ashr')
```
## Set the working directory and load the installed libraries
```r
setwd('/path/to/your/directory/with/dataset')
library(DESeq2)
library(EnhancedVolcano)
library(ashr)
```



```{r loading, include=FALSE}
setwd('C:/Users/suren/Google Drive Streaming/My Drive/test/cross_platform_backup/abi_deseq/abi_deseq/')
library(DESeq2)
library(EnhancedVolcano)
library(ashr)
```

# Input data generation
## Read the read counts matrix
As input, the DESeq2 package expects count data as obtained, e.g., from RNA-seq or another high-throughput sequencing experiment, in the form of a matrix of integer values. The value in the i-th row and the j-th column of the matrix tells how many reads can be assigned to gene i in sample j.
```{r counts}
counts_path <- "dataset/STAR_counts.tsv"
counts <- read.csv(
  counts_path, header = TRUE, sep = "\t", row.names = 1
)
dim(counts)
head(counts)
```

## Read the sample metadata
Information about each sample (e.g., subtype, condition) necessary to define groups for comparison.
```{r metadata}
metadata_path <- "dataset/metadata.tsv"
metadata <- read.csv(
  metadata_path, header = TRUE, sep = "\t", row.names = 1
)
dim(metadata)
head(metadata)
```


## Read gene annotation
Gene names and types to annotate results for interpretation.
```{r annotation}
gene_ann_path <- "dataset/gencode.v42.genes.tsv"
gene_ann <- read.csv(
  gene_ann_path, header = TRUE, sep = "\t", row.names = 1
)
dim(gene_ann)
head(gene_ann)
```


## Initialize DESeq2 object
The object class used by the DESeq2 package to store the read counts and the intermediate estimated quantities during statistical analysis is the DESeqDataSet.<br>
With the counts matrix and sample metadata, we can construct a DESeqDataSet:
```{r deseq}
dds <- DESeqDataSetFromMatrix(
  countData = counts,
  colData = metadata,
  design = ~ Subtype
)

```
## Pre-filtering
While it is not necessary to pre-filter low count genes before running the DESeq2 functions, there are two reasons which make pre-filtering useful:<br>
by removing rows in which there are very few reads, we reduce the memory size of the dds data object, <br>
and we increase the speed of count modeling within DESeq2.
```{r pre-filtering}
smallestGroupSize <- 3
keep <- rowSums(counts(dds) >= 10) >= smallestGroupSize
dds <- dds[keep,]
```


## DESeq2 normalization (median of ratios)
The size factor for each sample adjusts for differences in sequencing depth and RNA composition. The method used is median-of-ratios:<br>
1. Compute the geometric mean of each gene across all samples.<br>
2. For each gene in each sample, compute the ratio of the raw count to the geometric mean.<br>
3. For each sample, the median of these ratios becomes its size factor.<br>
For example, If a size factor is 0.8, the library is smaller than the reference, so counts will be scaled up.

```{r normalization}
dds <- estimateSizeFactors(dds)
sizeFactors(dds)
norm_counts <- counts(dds, normalized = TRUE)

```

## Use normalized counts to plot expression of gene ESR1 across subtypes

```{r esr1}
gene_name <- "ESR1"
gene_id <- rownames(gene_ann)[gene_ann$gene_name == gene_name]
boxplot(
  norm_counts[gene_id, ] ~ metadata[, "Subtype"],
  xlab = "Subtype", ylab = "Normalized counts",
  main = gene_name
)

```

# Differential expression analysis
The standard differential expression analysis steps are wrapped into a single function, DESeq.

```{r diffexp}
dds <- DESeq(dds)
```

## Factor levels
"Subtype" is a factor with multiple levels. Results tables are generated using the function results, which extracts a results table with log2 fold changes, p values and adjusted p values. With no additional arguments to results, the log2 fold change and Wald test p value will be for the last variable in the design formula, and if this is a factor, the comparison will be the last level of this variable over the reference level By default, R will choose a reference level for factors based on alphabetical order. Then, if you never tell the DESeq2 functions which level you want to compare against (e.g. which level represents the control group), the comparisons will be based on the alphabetical order of the levels.

We can do pairwise comparisons (contrasts)
```{r pairwise}
res_raw <- results(dds, contrast = c("Subtype", "LuminalA", "TNBC"))
res_raw <- res_raw[order(res_raw$padj), ]
```
The moderated log fold changes use a normal prior distribution, centered on zero and with a scale that is fit to the data. The shrunken log fold changes are useful for ranking and visualization, without the need for arbitrary filters on low count genes.
Shrinkage of effect size (LFC estimates) is useful for visualization and ranking of genes. To shrink the LFC, we pass the dds object to the function lfcShrink.
```{r shrinkage}
res_shrink <- lfcShrink(dds, res = res_raw, type = "ashr")
```

## Add gene annotation to results
```{r annotate}
res_raw <- cbind(res_raw, gene_ann[rownames(res_raw), ])
res_shrink <- cbind(res_shrink, gene_ann[rownames(res_shrink), ])
```

### TODO:
Make box plots (using normalized counts) for the 4 strongest DEGs (by adjusted p-value).
```{r assignment0}

```
## Volcano plots
### Before Shrinkage
```{r raw}
EnhancedVolcano(
  res_raw, x = "log2FoldChange", y = "padj",
  lab = res_raw$gene_name,
  pCutoff = 0.05, FCcutoff = 1,
  xlim = c(-10, 10), ylim = c(0, 30)
)
```

### After Shrinkage
```{r shrink}
EnhancedVolcano(
  res_shrink, x = "log2FoldChange", y = "padj",
  lab = res_shrink$gene_name,
  pCutoff = 0.05, FCcutoff = 1,
  xlim = c(-10, 10), ylim = c(0, 30)
)
```


## Filtering
Typical filters for focusing on biologically relevant and robust DEGs.
```{r significants}
res_signif <- na.omit(res_shrink)
res_signif <- res_signif[
  (res_signif$padj < 0.05) &
  (abs(res_signif$log2FoldChange) >= 1) &
  (res_signif$baseMean >= 100) &
  (res_signif$gene_type == "protein_coding"),
]
res_signif_up <- res_signif[res_signif$log2FoldChange > 0, ]
res_signif_down <- res_signif[res_signif$log2FoldChange < 0, ]
```

# Exporting results
This allows integration with pathway analysis, annotation tools, or publication preparation.
```{r exporting}
write.csv(as.data.frame(res_signif_up), 
          file="overexpressed_genes.csv")
write.csv(as.data.frame(res_signif_down), 
          file="downexpressed_genes.csv")
```


# Assignments
1. Make a scatter plot with X-axis as raw fold change and Y-axis as shrunken fold change. Color the points by baseMean.
```{r assignment1}

```

2. Repeat the analysis for a different contrast. For example, luminal A vs luminal B.
```{r assignment2}

```
